### ISSUE TITLE
[BUG] reduce docs

### ISSUE THREAD

**Issue by DenisYaroshevskiy** [jfalcou/eve#928]
* I believe operation needs to be commutative, not only associative:

```
[  a       b     c      d  ]
+
[  c       d     a      b  ]
[ a + c  b + d c + a d + b ]
```

* No mention of `splat(reduce)`
* No mention of default eve::plus.

**Comment by jtlap** [comment]
I agree, and commutativity is a real property that can  be satisfied. 
Moreover, I think we will need (but not now) thinks as ordered(reduce) and kahan(reduce) when as in floating case associativity is a pure wish (never realized because of rounding) and accuracy cannot be neglected

**Comment by jfalcou** [comment]
Fixed that and a bunch of other things in 64305f7e88fb7fe95fd80520ec2f7e25c0ed268a
